### ISSUE TITLE
Make `:Git branch` view interactive when changing branches with `coo`

### ISSUE THREAD

**Issue by kristoferfannar** [tpope/vim-fugitive#2234]
Warning, this is quite petty, but I'll add this here nevertheless.

I've been using `coo` to checkout to different branches with the branch view provided by `:Git branch`. It's very neat, but I'd like to see the asterisk move to be in front of the newly checkout-ed branch, after checking out to it.

Example:

I'm on main, as can be seen by the asterisk.

<img width="400" alt="image" src="https://github.com/tpope/vim-fugitive/assets/68929760/aafb8ac3-3c51-447f-b4ed-43b4463bb074">

Now, I've checked out to the *maxConnections* branch using `coo` within the `:Git branch` view, but the asterisk hasn't moved.

<img width="420" alt="image" src="https://github.com/tpope/vim-fugitive/assets/68929760/205aa746-76e0-4a1d-97c3-34fd6bd68dd1">

As I said, this is very petty and not too important, but it would definitely improve the user experience using the `:Git branch` command.

**Comment by wjt-outpost** [comment]
Agreed -- It would be quite nice to have a more dynamic `:Git branch` pane.
